src/sync/sync_permission.cpp

```diff
@@ -170,16 +171,22 @@ void Permissions::set_permission(std::shared_ptr<SyncUser> user,
                                  PermissionChangeCallback callback,
                                  const ConfigMaker& make_config)
 {
+    using namespace std::chrono;
     const auto realm_url = user->server_url() + permission.path;
     auto realm = Permissions::management_realm(std::move(user), make_config);
     CppContext context;
 
+    // Get the current time.
+    int64_t ns_since_epoch = duration_cast<nanoseconds>(system_clock::now().time_since_epoch()).count();
```

Comment:
Since there's apparently no portable standardized C++ way to get the epoch for the system clock, should I just add an extension point that can call out to the binding for the time since UNIX epoch?